stdlib/test/lit.cfg.py

```diff
@@ -47,7 +47,8 @@
 # Substitute %mojo for just `mojo` itself
 # since we're not supporting `--sanitize` initially
 # to allow running the tests with LLVM sanitizers.
-config.substitutions.insert(0, ("%mojo", "mojo"))
+config.substitutions.insert(0, ("%mojo", "mojo -D MOJO_ENABLE_ASSERTIONS"))
```

Comment:
**Thought** I'd like us to go a different direction, if possible.  I would like the stdlib to build and all tests pass in both modes (`mojo` and `mojo -D MOJO_ENABLE_ASSERTIONS`).  I recommend tackling this from the CI matrix level and plumbing the variable `(-D MOJO_ENABLE_ASSERTIONS`) from the matrix into the `lit.cfg.py` here which feeds into the tests.  This is more generic/composes better with other knobs in the future (think sanitizers and the like).  Thoughts?

Reply:
I don't mind doing this. I believe we'd need to document the rationale as it's not obvious why we need to run the tests twice. Is there a situation where a test wouldn't fail with assertions enabled, but would fail with assertions disabled?

Reply:
That is a valid point regarding running the tests twice and what that does to CI time.  I'm slightly paranoid about a case where our standard library only works with assertions enabled and not without (since without assertions is the default and common case for ends users/non-contributors).  One case is an assert that has side-affects (which is bad practice and you shouldn't do, of course, but I've seen these sort of things way too many times in C++ code).

I could be convinced to only run with assertions, but curious to get others input (CC: @laszlokindrat @rparolin).

Reply:
I'm not too worried about CI time as it can be two jobs running at the same time. I'm just wondering if it's worth the complexity.

> We can always err on the side of caution and run it twice in the CI. We can make running with assertions the default so that the contributors don't run all the tests twice locally, I don't want it to affect the DX. Let's wait for other inputs.

Reply:
This is on my list to enable internally ASAP and then we can adopt this PR.  It's a bit complicated since use different lit configs etc internally.  I'll sort it all out though.